### ISSUE TITLE
Logstash should log if confronted with unknown flags

### ISSUE THREAD

**Issue by stemperps** [elastic/logstash#11632]
We've encountered an interesting problem with our logstash pipelines: The pipeline takes a different action depending on a flag set in a column in a database table. The different values were predefined, but to our surprise a new flag was introduced without us knowing by a different team. Logstash could not process entries with that new flag (obviously), because it didn't know what to do. However, identifying the issue could have been easier if there had been a log message telling us that it didn't recognize the flag ... there wasn't, just a fairly generic failed request, so we had to dig a little deeper than we liked. 

In summary, it would be nice if Logstash could have a log message for that.

The version we're currently using is 7.5.2

**Comment by kares** [comment]
Hello Simon, "pipeline takes a different action" is a bit vague - an example conf might be useful.

**Comment by stemperps** [comment]
Hi Karol,

the failing part of the conf was this, where gsub didn't log it encountered an unknown flag:

```
filter
{
    mutate
    {
        id => "data_preparation"
        gsub => [
                    "foo", "I", "index",
                    "foo", "U", "update",
                    "foo", "D", "delete"
                ]
    }
}
```

FYI, we 'fixed' the problem (for the moment) with a regular expression.
